src/prototype_creep_resources.js

```diff
@@ -182,10 +182,8 @@ Creep.prototype.handleUpgrader = function() {
   say(this);
   this.spawnReplacement(1);
   var room = Game.rooms[this.room.name];
-  if (room.memory.attackTimer > 50 && room.controller.level > 6) {
-    if (room.controller.ticksToDowngrade > 10000) {
-      return true;
-    }
+  if ((room.energyAvailable < room.energyCapacityAvailable / 4 || (room.controller.level > 6 && room.memory.attackTimer > 50)) && room.controller.ticksToDowngrade > 10000) {
```

Comment:
this part block upgrader if not enouth energy for allow room to try to solve if there is a problem because of missing energy (acually doing alot of test for trying to solve this, it's sure there is a better way to do it ^^') which something can slow down a room or stop it completly);